slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -182,6 +183,30 @@ trait MySQLProfile extends JdbcProfile { profile =>
       expr(n)
       if(o.direction.desc) b" desc"
     }
+
+    // Overrides Slick's delete builder in order to produce a more efficient
+    // DELETE query for MySQL.
+    //
+    // Slick cannot directly handle multi-table DELETEs, i.e., using JOIN or
+    // USING, but it can handle subqueries in the WHERE clause of a DELETE.
+    // This is good except for the fact that MySQL doesn't know how to
+    // optimize such semi-join subqueries to joins in single-table DELETE
+    // queries. However, if the DELETE query is a multi-table DELETE, even if
+    // on a single table, then something in MySQL kicks in and optimizes the
+    // subquery to a more efficient JOIN. Further reading:
+    //
+    // - http://mysqlserverteam.com/multi-table-trick
+    // - https://mariadb.com/kb/en/mariadb/semi-join-subquery-optimizations
+    //
+    override def buildDelete: Result =
+      super.buildDelete match {
+        case Result(sql, setter) =>
+          // The replacement simply adds the table name after FROM in between
+          // DELETE and FROM. This makes it a multi-table DELETE query, but
+          // on a single table.
+          val s = sql.replaceFirst("""delete from\s+([^\s]+)""", "delete $1 from $1")
+          Result(s, setter)
+      }
```

Comment:
sounds great :). Instead of patching the query with search and replace, how about providing a proper override hook here?

Basicall we could pull out these three lines into a method `def buildDeleteFrom` that takes the table name.
https://github.com/slick/slick/blob/56877e93a4d28c7d05c78b3918e94eb81aed7412/slick/src/main/scala/slick/jdbc/JdbcStatementBuilderComponent.scala#L480-L482

And override it in MySQLProfile.scala